hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/jdbc/JdbcSqlConnector.java

```diff
@@ -291,14 +292,13 @@ private static SqlDialect resolveDialect(JdbcTable table, DagBuildContext contex
     }
 
     private static SqlDialect resolveDialect(DatabaseMetaData databaseMetaData) throws SQLException {
-        switch (databaseMetaData.getDatabaseProductName().toUpperCase(Locale.ROOT).trim()) {
-            case "MYSQL":
-                return new HazelcastMySqlDialect(SqlDialects.createContext(databaseMetaData));
-            case "MICROSOFT SQL SERVER":
-                return new HazelcastMSSQLDialect(SqlDialects.createContext(databaseMetaData));
-
-            default:
-                return SqlDialectFactoryImpl.INSTANCE.create(databaseMetaData);
+        String databaseProductName = DiscoverDatabase.getDatabaseProductName(databaseMetaData);
```

Comment:
IMHO will be better if `getDatabaseProductName` instead was `getDatabaseType` and returned enum - SQL_SERVER, MYSQL, POSTGRES, OTHER.